lib/createExplorer.js

```diff
@@ -74,11 +71,9 @@ module.exports = function (options) {
       })
       .then((result) => {
         if (result) return result;
-
-        const nextDirectory = joinPath(splitPath(directory).slice(0, -1));
-        if (!nextDirectory || directory === options.stopDir) return null;
-
-        return searchDirectory(nextDirectory);
+        if (!splitSearchPath) splitSearchPath = splitPath(directory);
+        if (!splitSearchPath.pop() || directory === options.stopDir) return null;
```

Comment:
I do not like relying on a side-effect within a condition. It's "efficient" in the sense of reducing lines of code — but that's not the goal. It's much less clear to read, and when I come back to this a year from now and need to remind myself of what's going on, it will be easier to do so with the side-effect in a condition.